src/components/ModiteList/index.tsx

```diff
@@ -38,13 +39,29 @@ let minutes: number; // used by tick
 let lastFilter: string = ''; // used by onFilter
 
 let lastScrollOffset = 0;
-const onScroll = ({ scrollOffset }: { scrollOffset: number }) => {
-  lastScrollOffset = scrollOffset;
-};
 
 const ModiteList: FunctionComponent<ModiteListProps & RouteComponentProps> = () => {
   const [modites, setModites]: [Modite[], React.Dispatch<any>] = useContext(ModitesContext);
   const [filter, setFilter]: [string, React.Dispatch<any>] = useState('');
+  const mapWindowRef: React.MutableRefObject<null> = useRef(null);
+  const searchBarWrapRef: React.MutableRefObject<null> = useRef(null);
+  const searchbarSpacerRef: React.MutableRefObject<null> = useRef(null);
+
+  const onScroll = ({ scrollOffset }: { scrollOffset: number }) => {
+    const threshold = 10; // scroll threshold to hit before acting on the layout
+
+    if ((lastScrollOffset < threshold && scrollOffset > threshold) || (lastScrollOffset > threshold && scrollOffset < threshold)) {
+      const modify = scrollOffset > threshold;
+
+      requestAnimationFrame(() => {
+        (mapWindowRef.current as any).style.height = modify ? 0 : '40vh';
```

Comment:
This code is very imperative which is in contrast with React's reactivity. I would consider setting state instead of styles. E.g. set `belowTheFold` state to true in order to apply certain CSS classes with the help of `classnames` npm module.

Reply:
fixed